src/Engine/Shaders/DiffuseShader.cpp

```diff
@@ -84,13 +92,20 @@ void DiffuseShader::render(const CameraComponent * camera, const Vector<Componen
     loadFloat(getUniform("silAngle"), silAngle);
     loadFloat(getUniform("numCells"), (float)numCells);
     loadInt(getUniform("cellIntensities"), cellIntensitiesTexture);
+    loadInt(getUniform("cellDiffuseScales"), cellDiffuseScalesTexture);
+    loadInt(getUniform("cellSpecularScales"), cellSpecularScalesTexture);
+
+    // TODO : move cell intensities and scales to material and initialize it during json pass
+    // TODO : only upload this data once when the material is loaded in
```

Comment:
@zackragozzino Can we set number of cells, cell intensities, diffuse scales, and specular scales through json? I can sit down with you to further explain the necessary data/options.